src/gt4py/storage/cartesian/interface.py

```diff
@@ -339,6 +339,9 @@ def from_array(
         dimensions=dimensions,
     )
 
-    storage[...] = storage_utils.asarray(data)
+    if cp is not None and isinstance(storage, cp.ndarray):
+        storage[...] = cp.asarray(data)
+    else:
+        storage[...] = np.asarray(data) 
```

Comment:
```python
device = layout.from_name(backend).device
# alternative:
# device = "gpu" if storage.__class__.__module__ = "cupy" else "cpu"
storage[...] = storage_utils.asarray(data, device=device)
```

Reply:
Thanks! I wanted that solution but was too tired to look for it.